compiler/bytecode/instruction.go

```diff
@@ -93,11 +93,29 @@ type InstructionSet struct {
 	isType       string
 	Instructions []*Instruction
 	count        int
-	argTypes     []int
+	argTypes     *ArgSet
+}
+
+type ArgSet struct {
```

Comment:
@shuheiktgw And we will have this `ArgSet` as argument's metadata instead of just `[]int`